src/range.rs

```diff
@@ -392,6 +394,20 @@ where
     }
 }
 
+impl<T> TryFrom<Range<T>> for TextRange
+where
+    T: TryInto<TextSize>,
+{
+    type Error = <T as TryInto<TextSize>>::Error;
+    #[inline]
+    fn try_from(value: Range<T>) -> Result<Self, Self::Error> {
+        Ok(TextRange::new(
+            value.start.try_into()?,
+            value.end.try_into()?,
+        ))
```

Comment:
Should check that start <= end?

Reply:
That would require me to rethink the error type and I didn't want to bother with that 😬 I guess discarding the `TryInto<TextSize>` error and replacing it with a generic one here would be probably fine